addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/PolicyDefinitionUnitTest.java

```diff
@@ -279,4 +288,67 @@ void shouldWarnIfRulesNotList() {
         assertThat(
                 progress.getWarnings().get(0), is(equalTo("!automation.error.options.badlist!")));
     }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "  defaultStrength: \n" + "  defaultThreshold: \n" + "  rules: ",
+                "defaultStrength:",
+                "defaultThreshold:"
+            })
+    void shouldReturnPolicyWithDefaultsIfDefinitionYamlContainsUndefinedStrengthThreshold(
+            String defnYamlStr) {
+        // Given
+        AutomationProgress progress = new AutomationProgress();
+        Yaml yaml = new Yaml();
+        Object data = yaml.load(defnYamlStr);
+
+        // When
+        policyDefinition.parsePolicyDefinition(data, "test", progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        ScanPolicy policy = policyDefinition.getScanPolicy("test", progress);
+        assertThat(policy, is(notNullValue()));
+        assertThat(policy.getDefaultStrength(), is(equalTo(AttackStrength.MEDIUM)));
+        assertThat(policy.getDefaultThreshold(), is(equalTo(AlertThreshold.MEDIUM)));
+        List<Plugin> rules = policy.getPluginFactory().getAllPlugin();
+        assertValueAppliedToRules(
+                () -> rules.get(0).getAttackStrength(),
+                () -> rules.get(rules.size() - 1).getAttackStrength(),
+                AttackStrength.MEDIUM);
+        assertValueAppliedToRules(
+                () -> rules.get(0).getAlertThreshold(),
+                () -> rules.get(rules.size() - 1).getAlertThreshold(),
+                AlertThreshold.MEDIUM);
+    }
+
+    @SuppressWarnings("rawtypes")
+    void assertValueAppliedToRules(Supplier<?> first, Supplier<?> last, Enum expected) {
```

Comment:
Why not pass the values directly? I'd have extracted the checks per plugin rather than try match both at the same time.
The enum can be parameterized.